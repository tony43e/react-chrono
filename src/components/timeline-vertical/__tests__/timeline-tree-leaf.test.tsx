import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Leaf from '../timeline-tree-leaf';

test('Check Timeline Leaf render', () => {
  const onClick = jest.fn();
  const onActive = jest.fn();

  render(
    <Leaf
      id={'tree-leaf'}
      className="random-class"
      onClick={onClick}
      onActive={onActive}
      active={false}
    />,
  );

  const element = screen.getByTestId('tree-leaf');
  expect(element).toBeInTheDocument();
});

test('Check Timeline Leaf snapshot', () => {
  const onClick = jest.fn();
  const onActive = jest.fn();

  render(
    <Leaf
      id={'tree-leaf'}
      className="random-class"
      onClick={onClick}
      onActive={onActive}
      active={false}
    />,
  );

  const element = screen.getByTestId('tree-leaf');
  expect(element).toMatchSnapshot();
});

test('Check Timeline Leaf Class & Styles', () => {
  const onClick = jest.fn();
  const onActive = jest.fn();

  render(
    <Leaf
      id={'tree-leaf'}
      className="random-class"
      onClick={onClick}
      onActive={onActive}
      active
    />,
  );

  const element = screen.getByTestId('tree-leaf');
  expect(element).toHaveClass('random-class');
  expect(element.children[0]).toHaveClass('random-class');
  expect(element.querySelector('.active')).toHaveClass('active');
});

test('Check Timeline Leaf onClick', async () => {
  const onClick = jest.fn();
  const onActive = jest.fn();

  render(
    <Leaf
      id={'tree-leaf'}
      className="random-class"
      onClick={() => onClick('zz12xx')}
      onActive={onActive}
      active
    />,
  );

  const element = screen.getByTestId('tree-leaf');
  fireEvent.click(element.children[0]);
  expect(onClick).toBeCalled();
  expect(onClick).toBeCalledWith('zz12xx');
  expect(onActive).toBeCalled();
});
