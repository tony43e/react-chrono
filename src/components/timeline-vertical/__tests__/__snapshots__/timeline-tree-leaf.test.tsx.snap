// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Timeline Leaf snapshot 1`] = `
<div
  class="random-class css-ey985k"
  data-testid="tree-leaf"
>
  <div
    class="random-class css-kooiip"
    data-testid="tree-leaf-click"
  >
    <div
      class="in-active css-1yzc5y5"
    />
  </div>
</div>
`;
